ddev/ddev.com
# Spruce up RSS feed

## The Issue

#140 and #141 raised some issues with the RSS feed:

1. Item links and GUIDs have had invalid URLs since the upgrade to Astro 3. (Courtesy of my oversight.)
2. Post content may include relative links that carry through to the feed.
3. The Feedly experience includes duplicate posts because of item 1.
4. The Feedly experience is underwhelming without post images and an icon. 

## How This PR Solves The Issue

I previously made a [quick change](https://github.com/ddev/ddev.com/commit/80476d0c976ebd7af48c88c4d059c56b2f38e522) to fix those invalid URLs (item 1).

This PR ensures that relative URLs in content are automatically converted to absolute ones (item 2).

The PR also ensures that feature images are included in the post content, which Feedly will use. It also adds a namespace just for Feedly in order to include an icon, logo, accent color, and generic cover image. All of this should amount to [nicer presentation in Feedly](https://blog.feedly.com/10-ways-to-optimize-your-feed-for-feedly/)—I tested to confirm what I could without the changes being live on ddev.com.

I don’t know if it’s possible to ask Feedly to ignore those bad URLs it previously got from the feed, but those are gone so the root cause is at least solved. Again, sorry for that being an issue!

## Manual Testing Instructions

You can add the RSS feed URL to Feedly to see it use feature images, but I think the reference logo and icons will need to exist on ddev.com to come through.
